src/redux/books/booksSlice.js

```diff
@@ -1,45 +1,85 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const client = axios.create({
+  baseURL: 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi',
+});
+
+export const createAppId = createAsyncThunk(
+  'books/createAppId',
+  async (name, thunkAPI) => {
+    try {
+      const resp = await client.post('/apps/');
+      return resp.data;
+    } catch (error) {
+      return thunkAPI.rejectWithValue('app not created');
+    }
+  },
+);
+export const getBooks = createAsyncThunk(
+  'books/getBooks',
+  async (name, thunkAPI) => {
+    try {
+      const resp = await client.get('apps/JWCmt9iaJ4Qaad541Qcl/books');
+
+      return resp.data;
+    } catch (error) {
+      return thunkAPI.rejectWithValue('something went wrong');
+    }
+  },
+);
+
+export const createNewBook = createAsyncThunk(
+  'books/createNewBook',
+  async (name, thunkAPI) => {
+    try {
+      const resp = await client.post('/apps/JWCmt9iaJ4Qaad541Qcl/books', name);
+      window.location.reload();
+      return resp.data;
+    } catch (error) {
+      return thunkAPI.rejectWithValue('something went wrong');
+    }
+  },
+);
+export const removeBook = createAsyncThunk(
+  'books/removeBook',
+  async (bookId, thunkAPI) => {
+    try {
+      const resp = await client.delete(`/apps/JWCmt9iaJ4Qaad541Qcl/books/${bookId}`, bookId);
+      window.location.reload();
```

Comment:
- [ ] Hi @ernestmusong, let us note that the purpose of using React is to build a single-page application (SPA), this entails that the browser does not reload. Reloading the browser defeats the purpose of this project and the subsequent ones. Kindly remove the reloading codes on lines 37 and 49.

  Then refactor your code so that the books can be added and deleted without reloading the browser page. Depending on how you wish to go about it one option is that you can add the book to the state if the response from the API is successful, and delete the book from the state using the id if the response of the API is successful.

Reply:
> * [ ]  Hi @ernestmusong, let us note that the purpose of using React is to build a single-page application (SPA), this entails that the browser does not reload. Reloading the browser defeats the purpose of this project and the subsequent ones. Kindly remove the reloading codes on lines 37 and 49.
>   Then refactor your code so that the books can be added and deleted without reloading the browser page. Depending on how you wish to go about it one option is that you can add the book to the store if the response from the API is successful, and delete the book from the store using the id if the response of the API is successful.

Thanks for your feedback @DammyShittu . I have now implemented the above changes.